File: java_old/game-online/game_service/src/main/java/ru/atom/gameservice/controller/GameServiceController.java
package ru.atom.gameservice.controller;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import ru.atom.gameservice.GameServer;
import ru.atom.gameservice.GameSession;

@Controller
@RequestMapping("game")
public class GameServiceController {

    private static final Logger logger = LoggerFactory.getLogger(GameServiceController.class);
    private GameServer gameServer;

    @Autowired
    public GameServiceController(GameServer gameServer) {
        this.gameServer = gameServer;
    }

    @RequestMapping(path = "create",
            method = RequestMethod.POST,
            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
    @ResponseStatus(HttpStatus.OK)
    public ResponseEntity<Long> create(@RequestParam("count") int count) {
        long id = gameServer.getNextId();
        gameServer.addGameSession(String.valueOf(id), new GameSession(count));
        logger.info("Game {} created", id);
        return ResponseEntity.ok(id);
    }

    @RequestMapping(path = "start",
            method = RequestMethod.POST,
            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
    @ResponseStatus(HttpStatus.OK)
    public ResponseEntity start(@RequestParam("gameId") long gameId) {
        logger.info("Game {} started", gameId);
        new Thread(gameServer.getGameSessionByName(String.valueOf(gameId))).run();
        return ResponseEntity.ok().build();
    }
}

File: GameService/src/main/kotlin/io/rybalkinsd/kotlinbootcamp/http/GameServiceController.kt
package io.rybalkinsd.kotlinbootcamp.http

import io.rybalkinsd.kotlinbootcamp.game.GameRepository
import io.rybalkinsd.kotlinbootcamp.util.logger
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam

@Controller
@RequestMapping("/game")
class GameServiceController {
    val log = logger()

    @Autowired
    lateinit var gameRepository: GameRepository

    //  todo make with json everywhere
    @PostMapping(
            path = ["/create"],
            consumes = [MediaType.APPLICATION_FORM_URLENCODED_VALUE]
    )
    fun create(@RequestParam("playerCount") playerCount: Int):
            ResponseEntity<String> {
        log.info("Added request for $playerCount players")
        return when {
            playerCount < 2 -> ResponseEntity.badRequest().body("Too low player count")
            playerCount > 4 -> ResponseEntity.badRequest().body("Too high player count")
            else -> {
                // beanFactory create game and and handle errors
                val game = gameRepository.createGame(playerCount)
                ResponseEntity.ok(game.id.toString())
            }
        }
    }

    @GetMapping(
            path = ["/create"],
            consumes = [MediaType.ALL_VALUE]
    )
    fun createe(@RequestParam("playerCount") playerCount: Int):
            ResponseEntity<String> = when {
        playerCount < 2 -> ResponseEntity.badRequest().body("Too low player count")
        playerCount > 4 -> ResponseEntity.badRequest().body("Too high player count")
        else -> {
            // beanFactory create game and and handle errors
            val game = gameRepository.createGame(playerCount)
            ResponseEntity.ok(game.id.toString())
        }
    }

    //  todo make with json everywhere
    @RequestMapping(
            path = ["/start"],
            method = [RequestMethod.POST],
            consumes = [MediaType.APPLICATION_FORM_URLENCODED_VALUE]
    )
    fun start(@RequestParam("gameId") gameId: Int):
            ResponseEntity<String> {
        val game = gameRepository.getGame(gameId)
                ?: return ResponseEntity.badRequest().body("Unknown game id - $gameId")
        game.start()
        return ResponseEntity.ok().build()
    }
}
